File: RestaurantManager/Controllers/DishController.cs
using Restaurant.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace RestaurantManager.Controllers
{
    public class DishController : Controller
    {
        private RestaurantMenuContext _dbContext;

        public DishController()
        {
            _dbContext = new RestaurantMenuContext();
        }
        
        public ActionResult Index()
        {
            return View();
        }

        public JsonResult GetDishes()
        {
            return Json(_dbContext.Dishes.ToList(), JsonRequestBehavior.AllowGet);
        }

        public JsonResult GetDishById(int id)
        {
            return Json(_dbContext.Dishes.FirstOrDefault(r => r.Id == id), JsonRequestBehavior.AllowGet);
        }

        public JsonResult GetDishByName(string name)
        {
            return Json(_dbContext.Dishes.FirstOrDefault(r => r.Name == name), JsonRequestBehavior.AllowGet);
        }

        public JsonResult AddDish(Dish dish)
        {
            _dbContext.Dishes.Add(dish);
            _dbContext.SaveChanges();

            return GetDishes();
        }

        public JsonResult UpdateDish(Dish dish)
        {
            _dbContext.Entry(dish).State = System.Data.Entity.EntityState.Modified;
            _dbContext.SaveChanges();

            return GetDishes();
        }

        public JsonResult DeleteDish(int id)
        {
            var dish = _dbContext.Dishes.FirstOrDefault(r => r.Id == id);

            if (dish != null)
                _dbContext.Dishes.Remove(dish);
            
            _dbContext.SaveChanges();

            return GetDishes();
        }
    }
}

File: RestaurantManager/app/controllers/dish.controller.js
(function () {
    'use strict';

    angular
        .module('app')
        .controller('DishController', DishController);

    function DishController($scope, $location, $routeParams, DishDataService, RestaurantDataService) {
        
        getDishes();        

        RestaurantDataService.getRestaurants().then(function (restaurantRes) {
            $scope.restaurants = restaurantRes;
        });

        if ($routeParams && $routeParams.id)
            getDishById();

        $scope.dishService = {
            getDishes: getDishes,
            getDishById: getDishById,
            getDishByName: getDishByName,
            addDish: addDish,
            updateDish: updateDish,
            deleteDish: deleteDish
        };

        $scope.goAddDish = function () {
            $location.path('/AddDish/');
        };

        $scope.goDishList = function () {
            $location.path('/DishList/');
        }

        function addDish() {
            DishDataService
                 .addDish($scope.dish)
                 .then(function (dishes) {
                     $location.path('/DishList');
                     alert('Novo prato registrado com sucesso.');
                     $scope.dishes = dishes;
                     $scope.dish = {};
                 });
        }

        function deleteDish(id) {
            DishDataService
                .deleteDish(id)
                .then(function (dishes) {
                    $location.path('/DishList');
                    alert('Prato com Id: ' + id + ' deletado com sucesso');                    
                    $scope.dishes = dishes;
                });
        }

        function updateDish() {
            var req = {
                Id: $scope.dish.Id,
                Name: $scope.dish.Name,
                Price: $scope.dish.Price,
                RestaurantId: $scope.dish.RestaurantId
            };

            DishDataService
                .updateDish(req)
                .then(function (dishes) {
                    $location.path('/DishList');
                    alert('Alteração realizada com sucesso.');                    
                    $scope.dishes = dishes;
                });
        }

        function getDishes() {
            DishDataService
                .getDishes()
                .then(function (dishes) {
                    $scope.dishes = dishes;
                });
        }

        function getDishById(id) {
            DishDataService
                .getDishById(id || $routeParams.id)
                .then(function (dish) {
                    $scope.dish = dish;
                    $scope.restaurant = dish.Restaurant;
                });
        }

        function getDishByName(name) {
            DishDataService
                .getDishByName($routeParams.name)
                .then(function (dish) {
                    $scope.dish = dish;
                    $scope.restaurant = dish.Restaurant;
                });
        }
    }
})();
